core/blockchain.go

```diff
@@ -537,16 +537,17 @@ func (bc *BlockChain) ResetWithGenesisBlock(genesis *types.Block) error {
 	defer bc.chainmu.Unlock()
 
 	// Prepare the genesis block and reinitialise the chain
-	if err := bc.hc.WriteTd(genesis.Hash(), genesis.NumberU64(), genesis.Difficulty()); err != nil {
-		log.Crit("Failed to write genesis block TD", "err", err)
+	batch := bc.db.NewBatch()
```

Comment:
The change here is we can write the `td` and `block` of genesis into the single batch. 
While before the change, it writes the `td` first and then `block`. Not atomicity guarantee.

Regarding the `db` instance, these two refer to the same one(`bc` and `bc.hc`). But actually we might need. Since we need to read some data from both `bc` and `hc`